src/windows_registry.rs

```diff
@@ -218,6 +205,48 @@ mod impl_ {
         }
     }
 
+    /// Checks to see if the `VSCMD_ARG_TGT_ARCH` environment variable matches the
+    /// given target's arch. Returns `None` if the variable does not exist.
+    #[cfg(windows)]
+    fn is_vscmd_target(target: &str) -> Option<bool> {
+        let vscmd_arch = env::var("VSCMD_ARG_TGT_ARCH").ok()?;
+        // Convert the Rust target arch to its VS arch equivalent.
+        let arch = match target.split("-").next() {
+            Some("x86_64") => "x64",
+            Some("aarch64") => "arm64",
+            Some("i686") | Some("i586") => "x86",
+            Some("thumbv7a") => "arm",
+            // An unrecognized arch.
+            _ => return Some(false),
+        };
+        Some(vscmd_arch == arch)
+    }
+
+    /// Attempt to find the tool using environment variables set by vcvars.
+    pub fn find_msvc_environment(target: &str, tool: &str) -> Option<Tool> {
+        // Early return if the environment doesn't contain a VC install.
+        if env::var_os("VCINSTALLDIR").is_none() {
+            return None;
+        }
+        let vs_install_dir = env::var_os("VSINSTALLDIR")?.into();
+
+        // If the vscmd target differs from the requested target then
+        // attempt to get the tool using the VS install directory.
+        if is_vscmd_target(target) == Some(false) {
+            // We will only get here with versions 15+.
+            tool_from_vs15plus_instance(tool, target, &vs_install_dir)
```

Comment:
I think this should perhaps return `None` to preserve what the previous code did? That way the `.or_else` chain up above will fall through to the next branch?

Reply:
I wanted to try using the install directory given in the environment variable instead of probing for it. My assumption being that the user's intent was to use that instance of Visual Studio even though the given targets don't match up. But I can change it if that's not appropriate.

Reply:
Ah ok makes sense! Just wanted to double check, sounds like you know a lot more about these installs than I do!